python/pyarrow/_dataset.pyx

```diff
@@ -84,7 +84,25 @@ cdef CFileSource _make_file_source(object file, FileSystem filesystem=None):
 
 
 cdef class Expression(_Weakrefable):
-
+    """
+    A logical expression to be evaluated against some input.
+
+    To create an expression:
+     - Use the factory function ``pyarrow.dataset.scalar()`` to create a
+       scalar (not necessary when combined, see example below).
+     - Use the factory function ``pyarrow.dataset.field()`` to reference
+       a field (column in table).
+     - Compare fields and scalars with ``<``, ``<=``, ``==``, ``>=``, ``>``.
+     - Combine expressions using python operators ``&`` (logical and),
+       ``|`` (logical or) and ``~`` (logical not).
+       Note: python keywords ``and``, ``or`` and ``not`` do not work to combine expressions.
+
+    Examples:
+    --------
+    >>> from pyarrow.dataset import scalar, field
```

Comment:
In the other part of the docs, we use `import pyarrow.dataset as ds`, we should probably be consistent (https://arrow.apache.org/docs/python/dataset.html)